components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java

```diff
@@ -0,0 +1,389 @@
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    /**
+     * Sets username for basic auth plugin enabled servers
+     */
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    /**
+     * Sets password for basic auth plugin enabled servers
+     */
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public boolean isAutoCommit() {
+        return autoCommit;
+    }
+
+    /**
+     * If true, each producer operation will be automatically followed by a commit
+     */
+    public void setAutoCommit(boolean autoCommit) {
+        this.autoCommit = autoCommit;
+    }
+
+    public SolrClient getSolrClient() {
+        return solrClient;
+    }
+
+    /**
+     * Uses the provided solr client to connect to solr. When this parameter is not specified, camel applies the
+     * following rules to determine the SolrClient: 1) when zkHost or zkChroot (=zookeeper root) parameter is set, then
+     * the CloudSolrClient is used. 2) when multiple hosts are specified in the uri (separated with a comma), then the
+     * CloudSolrClient (uri scheme is 'solrCloud') or the LBHttpSolrClient (uri scheme is not 'solrCloud') is used. 3)
+     * when the solr operation is INSERT_STREAMING, then the ConcurrentUpdateSolrClient is used. 4) otherwise, the
+     * HttpSolrClient is used. Note: A CloudSolrClient should point to zookeeper endpoint(s); other clients point to
+     * Solr endpoint(s). The SolrClient can also be set via the exchange header 'CamelSolrClient'.
+     */
+    public void setSolrClient(SolrClient solrClient) {
+        this.solrClient = solrClient;
+    }
+
+    public HttpClient getHttpClient() {
+        return httpClient;
+    }
+
+    /**
+     * Sets the http client to be used by the solrClient. This is only applicable when solrClient is not set.
+     */
+    public void setHttpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public boolean getUseConcurrentUpdateSolrClient() {
+        return useConcurrentUpdateSolrClient;
+    }
+
+    void setUseConcurrentUpdateSolrClient(boolean useConcurrentUpdateSolrClient) {
+        this.useConcurrentUpdateSolrClient = useConcurrentUpdateSolrClient;
+    }
+
+    public SolrEndpoint getSolrEndpoint() {
+        return solrEndpoint;
+    }
+
+    void setSolrEndpoint(SolrEndpoint solrEndpoint) {
+        this.solrEndpoint = solrEndpoint;
+    }
+
+    public SolrConfiguration deepCopy() {
+        try {
+            return (SolrConfiguration) this.clone();
+        } catch (CloneNotSupportedException e) {
+            LOG.error(
+                    String.format(
+                            "Could not generate new configuration based on configuration of existing endpoint %s",
+                            getSolrEndpoint()),
+                    e);
```

Comment:
I think there's no need to call `String.format` here. Calling just the `LOG.error` should be just fine.

Reply:
fixed